tests/kernel/fifo/fifo_timeout/src/main.c

```diff
@@ -199,12 +199,14 @@ static int test_multiple_threads_get_data(struct timeout_order_data *test_data,
 
 		data = k_fifo_get(&timeout_order_fifo, K_FOREVER);
 
-		if (data->q_order == ii) {
+		if (data && data->q_order == ii) {
 			TC_PRINT(" thread (q order: %d, t/o: %d, fifo %p)\n",
 				data->q_order, data->timeout, data->fifo);
 		} else {
-			TC_ERROR(" *** thread %d woke up, expected %d\n",
+			if (data) {
```

Comment:
Why do we need this extra if statement?

Reply:
if data is NULL it will hit the else section, and if data->q_order is dereferenced, in the else case, without checking data, it will crash.